docs/content/docs/connectors/datastream/dynamodb.md

```diff
@@ -23,16 +23,140 @@ KIND, either express or implied.  See the License for the
+  .setStreamArn("arn:aws:dynamodb:us-east-1:1231231230:table/test/stream/2024-04-11T07:14:19.380")
+  .setSourceConfig(sourceConfig)
+  // User must implement their own deserialization schema to translate change data capture events into custom data types    
+  .setDeserializationSchema(dynamodbDeserializationSchema)
+  .build()
+
+val env = StreamExecutionEnvironment.getExecutionEnvironment()
+
+// Specify watermarking strategy and the name of the DynamoDB Streams Source operator.
+// Specify return type using TypeInformation.
+// Specify also UID of operator in line with Flink best practice.
+val cdcEventsWithEventTimeWatermarks = env.fromSource(dynamoDbStreamsSource, WatermarkStrategy.<String>forMonotonousTimestamps().withIdleness(Duration.ofSeconds(1)), "DynamoDB Streams source")
+  .uid("custom-uid")
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The above is a simple example of using the `DynamoDbStreamsSource`.
+- The DynamoDB Stream being read from is specified using the stream ARN.
+- Configuration for the `Source` is supplied using an instance of Flink's `Configuration` class.
+  The configuration keys can be taken from `AWSConfigOptions` (AWS-specific configuration) and `DynamodbStreamsSourceConfigConstants` (DynamoDB Streams Source configuration).
+- The example specifies the starting position as `TRIM_HORIZON` (see [Configuring Starting Position](#configuring-starting-position) for more information).
+- The deserialization format is as `SimpleStringSchema` (see [Deserialization Schema](#deserialization-schema) for more information).
+- The distribution of shards across subtasks is controlled using the `UniformShardAssigner`  (see [Shard Assignment Strategy](#shard-assignment-strategy) for more information).
+- The example also specifies an increasing `WatermarkStrategy`, which means each record will be tagged with event time specified using `approximateCreationDateTime`.
+  Monotonically increasing watermarks will be generated, and subtasks will be considered idle if no record is emitted after 1 second.
+
+### Configuring Starting Position
+
+To specify the starting position of the `DynamodbStreamsSource`, users can set the `DynamodbStreamsSourceConfigConstants.STREAM_INITIAL_POSITION` in configuration.
+- `LATEST`: read all shards of the stream starting from the latest record.
+- `TRIM_HORIZON`: read all shards of the stream starting from the earliest record possible (data is trimmed by DynamoDb after 24 hours).
+
+### Deserialization Schema
+
+The `DynamoDbStreamsSource` provides the `DynamoDbStreamsDeserializationSchema<T>` interface to allow users to implement their own
+deserialization schema to convert DynamoDB change data capture events into custom event types.
+
+The `DynamoDbStreamsDeserializationSchema<T>#deserialize` method takes in an instance of `Record` from the DynamoDB model.
+The `StreamRecord` will contain different fields, depending on the configuration of the DynamoDB Stream.
```

Comment:
what is this `StreamRecord`? It seems to be getting first time introduced

Reply:
This is the internal class for `Record`. I will rephrase to use `Record`. Good spot